pkg/nsx/services/subnet/subnet.go

```diff
@@ -157,13 +157,14 @@ func (service *SubnetService) CreateOrUpdateSubnet(obj client.Object, vpcInfo co
 
 			changed = common.CompareResource(SubnetToComparable(existingSubnet), SubnetToComparable(nsxSubnet))
 			if changed {
-				// Only tags and dhcp are expected to be updated
+				// Only tags, dhcp and advancedConfig are expected to be updated
 				// inherit other fields from the existing Subnet
 				// Avoid modification on existingSubnet to ensure
 				// Subnet store is only updated after the updating succeeds.
 				updatedSubnet := *existingSubnet
 				updatedSubnet.Tags = nsxSubnet.Tags
 				updatedSubnet.SubnetDhcpConfig = nsxSubnet.SubnetDhcpConfig
+				updatedSubnet.AdvancedConfig = nsxSubnet.AdvancedConfig
```

Comment:
I also prefer to update gateway_addresses and dhcp_server_address only to avoid potential overwriting some of the fields with the default value.

Reply:
done